Formula/hashlink.rb

```diff
@@ -0,0 +1,25 @@
+class Hashlink < Formula
+  desc "Virtual machine for Haxe"
+  homepage "https://hashlink.haxe.org/"
+  url "https://github.com/HaxeFoundation/hashlink.git",
+    :tag =>"1.11"
+
+  depends_on "cmake"
+  depends_on "jpeg-turbo"
+  depends_on "libogg"
+  depends_on "libpng"
+  depends_on "libuv"
+  depends_on "libvorbis"
+  depends_on "mbedtls"
+  depends_on "openal-soft"
+  depends_on "sdl2"
+
+  def install
+    system "make"
+    system "make", "install", "INSTALL_DIR=#{prefix}"
+  end
+
+  test do
+    system "hl", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
HL is a jit runtime, so none of the suggestions apply.  I'm also not seeing helpful examples in any of the more popular runtimes, like Lua.  What would be a good test for a bytecode interpreter?  I tried embedding bytecode in the ruby file with no luck.

Reply:
E.g., I could create a small "hello world" example, but it would need to be in a separate file, and I'm not seeing anything analogous to this approach in homebrew-core.

Reply:
You could download an example from somewhere using a resource block